lib/src/main/java/org/asamk/signal/manager/Manager.java

```diff
@@ -2365,9 +2365,9 @@ public String getContactOrProfileName(String number) {
 
         final var profileEntry = account.getProfileStore().getProfileEntry(address);
         if (profileEntry != null && profileEntry.getProfile() != null) {
-            return profileEntry.getProfile().getDisplayName();
+            String name = profileEntry.getProfile().getDisplayName();
+            return name==null?"":name;
```

Comment:
This null check should just be in the dbus layer.

Reply:
On the Dbus layer I can no longer distiguish if the null is due to emtpy name in profile or number not found/badly formatted.
Alternative would be to return null for empty and throw an exception for "not found".

Reply:
To iterate a bit more maybe on the case: My client program can display the list of contacts or group members. Here an empty profile name (which can happen e.g. if the account was created by signal-cli) should simply fall back to the phone number (like your writer that uses the same method).
However the client also tries to translate numbers to names and it can happen that this is done for an unknown or badly formatted number. In that case I would rather not return an empty string, but raise an error to the user.